### PR TITLE
Remove host header from author-api listener rules.

### PR DESCRIPTION
**Pull request by samiwel**
## What is the context of this PR?
Since the introduction of the author short url was done manually, outside of terraform, there were conflicting listener rule priority values upon the next deployment to prod.

This PR removes the host from the listener rules for the API so that traffic should be routed to the API both when the short url is used and also when the prod dns name is used.

**Edit: I have modified this PR to also (optionally) create a listener rule for the author short URL and redirect traffic from HTTP 80 to HTTPS 443.**

## How to test
Deploy into a dev environment and ensure that app behaves correctly.
Observe in the listener rules for the load balancer that the host header is not present when routing traffic to the API.

### CODE REVIEW

File: author.tf [hunk 1]
```diff
@@ -315,6 +315,7 @@ module "author-api" {
   ecs_alb_security_group     = ["${module.author-eq-ecs.ecs_alb_security_group}"]
   launch_type                = "FARGATE"
   alb_listener_path_patterns = ["/graphql*", "/launch*", "/status", "/export*", "/import*", "/signIn"]
+  aws_alb_use_host_header    = false
```
Comment (tgandrews) [inline author.tf]: Is this something the aws module provides or does it require another change?

Comment (samiwel) [reply author.tf]: This is a variable exposed by `eq-ecs-deploy` it decides whether the listener rule is created with the host attribute or not.

**Review by tgandrews — APPROVED** [review]

**Comment by tgandrews** [comment]
Worked for me - I could see the change before and after.

**Comment by samiwel** [comment]
@tgandrews Made a couple of extra changes since you tested. Might wanna have another look?

File: author.tf [hunk 2]
```diff
@@ -498,3 +499,41 @@ module "author-dynamodb" {
 output "author_service_address" {
   value = "${module.author.service_address}"
 }
+
+resource "aws_lb_listener_rule" "listener_rule" {
```
Comment (tgandrews) [inline author.tf]: Why is this needed?

Comment (samiwel) [reply author.tf]: It creates the LB listener rule to forward traffic destined for `author.ons.gov.uk` to Author's target group.

File: author.tf (hunk 1 above)
Comment (jonnyshaw89) [reply author.tf]: instead of passing a boolean, can we pass a wildcard eg `alb_host_header = "*"` to just override a default?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: author.tf (hunk 2 above)
Comment (tgandrews) [reply author.tf]: I'm probably being dumb here but isn't this covered by the other rules not requiring a specific host header?

Comment (samiwel) [reply author.tf]: The host headers were only removed for the API so the host header still exists on the listener rule for author's target group. I guess I could also remove the host headers for author and get rid of this rule but would need to adjust the priorities of the rules to ensure that author's rules get evaluated after everything else?

**Comment by samiwel** [comment]
I tested the wildcard method but didn't seem to work.

File: developer_defaults.tf [hunk 3]
```diff
@@ -209,3 +209,8 @@ variable "create_ecs_internal_elb" {
   description = "Deploy an internal load balancer for ECS"
   default     = false
 }
+
+variable "create_short_url_listener_rule" {
```
Comment (jonnyshaw89) [inline developer_defaults.tf]: Whats this?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: developer_defaults.tf (hunk 3 above)
Comment (samiwel) [reply developer_defaults.tf]: No longer required, will remove

**Review by jonnyshaw89 — APPROVED** [review]
